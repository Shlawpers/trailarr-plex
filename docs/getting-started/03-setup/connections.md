# Connections

Once your Profiles are set up to your liking, it's time to add your Radarr and Sonarr connections. This allows Trailarr to communicate with your *Arr instances, discover your media, and manage trailers.

![Connections](connections-list.png)

Setting up a connection involves three main steps:

## 1. **Connection Details**

- Navigate to `Settings > Connections`.
- Click on `Add New` button.
![Add Connection](connection-add.png)
- Fill in the basic details for your *Arr instance (e.g., Radarr or Sonarr):
    - **Name:** A friendly name for this connection (e.g., "Main Radarr", "4K Sonarr", or simply "Radarr").
      I am adding a `Radarr` connection, so I named it `Radarr`
    - **Arr Type:** Select Radarr or Sonarr.
      It's already set to `Radarr` by default, so I didn't change it.
    - **Monitor Type:** Choose which Media items Trailarr needs to monitor from this connection.
    - **Server URL:** The full URL of your *Arr instance. Eg: `http://192.168.0.15:6969`
        
        !!! warning "Do NOT use `localhost`"
            Do not use `localhost` as there are numerous problems with that and it might not always work!

            Instead use a static IP address
      
    - **API Key:** Your API key from Radarr/Sonarr (found in its `Settings > General` section).
- Once these details are filled in, the `Test` button will become available, click the button.
- Trailarr will attempt to connect to the *Arr API. Upon a successful connection:
    - The Arr API version will be displayed.
    - Trailarr will automatically fetch the root folders configured in that *Arr instance.

!!! warning ""
    If the connection fails, an error will be shown with details, correct them and click `Test` again!

## 2. **Path Mappings**

After a successful initial test, Trailarr needs to know how to access the media files managed by this *Arr instance. This is where you map the root folders reported by the *Arr API to the paths where Trailarr can find them.

!!! tip ""
    Remeber the `Volume Mappings` we set during [installation](../02-installation/docker-compose.md#explanation-of-options), we use them here!

![Connection - Add - Path Mappings](connection-add-path-mappings.png)

- For each root folder retrieved from the *Arr instance, you'll see a "Path From" and a "Path To" field.
    - **Path From (read only):** This is the root folder path as known by your *Arr instance (e.g., `/movies` or `/tv`). This field is usually read-only.
    - **Path To:** This is where you tell Trailarr to find the *contents* of that "Path From" folder *within Trailarr's own file system (i.e., inside its Docker container)*.
    
    !!! tip ""
        Use the üìÅ icon next to the text input field for `Path To`, a dialog will open showing files and folders *inside Trailarr container*, navigate to the folder for the `Path To` and click `Confirm`.
    
    !!! danger ""
        You can `Delete` a `Path Mapping` by clicking on the `X` button next to it!

          - You can remove it if your `Path From` is same as `Path To`.
          - You can leave them there even if `Path From = Path To`.
    
    We have some examples (the same ones from `Docker Compose`):

    ??? example "Examples 1 & 2 - TRaSH Guides - SAME path as `Radarr`/`Sonarr`"
        If you used a `volume mapping` that is SAME as `Radarr`/`Sonarr` in `Trailarr` then `Path To` is same as `Path To`:

        ```yaml hl_lines="7 14"
        services:
          radarr:
            image: ghcr.io/hotio/radarr:latest
            # Other options here
            volumes:
            - /docker/appdata/radarr:/config
            - /data:/data
            # Other options here
          trailarr:
            image: nandyalu/trailarr:latest
            # Other options here
            volumes:
            - /docker/appdata/trailarr:/config
            - /data:/data
            # Other options here
        ```
        OR

        ```yaml hl_lines="7 14"
        services:
          radarr:
            image: ghcr.io/hotio/radarr:latest
            # Other options here
            volumes:
              - /docker/appdata/radarr:/config
              - /mnt/movies_disk/media/movies:/media/movies
            # Other options here
          trailarr:
            image: nandyalu/trailarr:latest
            # Other options here
            volumes:
              - /docker/appdata/trailarr:/config
              - /mnt/series_disk/media/tv:/media/tv
            # Other options here
        ```
        The folder dialog might already in the folder you are looking for, just press `Confirm`!
    
    ??? example "Example 3 - DIFFERENT path than `Radarr`/`Sonarr`"
        If you used a `volume mapping` which is DIFFERENT than what you had in `Radarr`/`Sonarr`, meaning - we changed the actual location of where the media files are present, so we need to tell `Trailarr` where to find them:

        ```yaml hl_lines="5 9 13-14"
        # USE THIS INSTEAD
        radarr:
          # other config
          volumes:
            - /mnt/disk1/media/movies:/data
        sonarr:
          # other config
          volumes:
            - /mnt/disk2/media/tv:/data
        trailarr:
          # other config
          volumes:
            - /mnt/disk1/media/movies:/data/movies # <-- MODIFIED
            - /mnt/disk2/media/tv:/data/tv # <-- MODIFIED
        ```

        - The root folder for `Radarr` connection is `/data/movies` or under that folder!
        - The root folder for `Sonarr` connection is `/data/tv` or under that folder!

        Navigate to the folder and press `Confirm`!

        So, when Trailarr gets a Movie folder path it will use the `Path Mapping` to convert it like this

        ```yaml
        Radarr Media Path: '/data/The Matrix (1999)'
        Trailarr Media Path: '/data/movies/The Matrix (1999)'
        ```

        Trailarr knows where to find it, so all will work as expected!

    ??? example "Example 4 - Windows users"

        We used `volume mappings` like this for `Trailarr`:

        ```yaml hl_lines="6-9"
        services:
          trailarr:
            # Other options here
            volumes:
              - trailarr_data:/config # Docker volume for appdata, the first part `trailarr_data` is the volume name
              - M:\Movies:/m/movies   # Movies drive
              - R:\TV:/r/tv           # TV series drive 1
              - S:\TV:/s/tv           # TV series drive 2
              - T:\TV:/t/tv           # TV series drive 3
            restart: unless-stopped

        volumes:
          trailarr_data:  # volume name, should match the volume name in the service
            # Any extra options for the volume if needed
        ```

        - The root folder for `Radarr` connection is `/m/movies` or under that folder!
        - The root folders for `Sonarr` connection are:
            - `/r/tv`
            - `/s/tv`
            - `/t/tv`

        Navigate to the appropriate folder and press `Confirm`!
    
    ??? example "Example 5 - Non-Docker `Radarr`/`Sonarr`"
        Since `Radarr` and `Sonarr` are Non-Docker installations, we don't have any `volume mappings` for them, so we created some, now we need to tell `Trailarr` where to find them!

        If we used something like this

        ```yaml hl_lines="6"
        services:
          trailarr:
            # Other options here
            volumes:
              - /docker/appdata/trailarr:/config # appdata
              - /mnt/disk1/movies:/media/movies # Movies Folder
              - /mnt/disk2/tv:/media/tv # TV Folder
            restart: unless-stopped
            # Other options here
        ```

        - The root folder for `Radarr` connection is `/media/movies` or under that folder!
        - The root folder for `Sonarr` connection is `/media/tv` or under that folder!

        Navigate to the folder and press `Confirm`! 

        So, when Trailarr gets a Movie folder path it will use the `Path Mapping` to convert it like this

        ```yaml
        Radarr Media Path: '/mnt/disk1/movies/The Matrix (1999)'
        Trailarr Media Path: '/media/movies/The Matrix (1999)'
        ```
        
        Trailarr knows where to find it, so all will work as expected!

Once all Path Mappings are filled out, you should have something like below, you can move on to next step!

![Connections - Add - Final Paths](connection-add-path-final.png)

!!! tip ""
    Crucially, ensure that each "Path From" is correctly mapped to its corresponding location in Trailarr's file system.

## 3. **Validate & Submit**

- Once you have updated the "Path To" for all root folders, the `Test` button will become enabled again.
- Click this button. Trailarr will now attempt to validate these path mappings by checking if it can access them.
- The server will return a result indicating success or failure.

    ![Connection - Add - Ready to Submit](connection-add-path-submit.png)

- If all mappings are validated successfully, the `Test` button will change to `Submit`, click it to save the connection.

You have now successfully added an *Arr connection! Trailarr will begin to sync media information from this instance. ü•≥üéâ

You can repeat the same steps for other `Radarr`/`Sonarr` connections you want `Trailarr` to monitor!

We are almost done with the seup, just read some information and tips we put together for you in [Next Steps](../04-next-steps.md).