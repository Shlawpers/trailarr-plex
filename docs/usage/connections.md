## Add Radarr/Sonarr Connections

1. Navigate to the application in your browser at [http://localhost:7889/settings/connections](http://localhost:7889/settings/connections){:target="_blank"}.

    !!! info "Trailarr Login"
        When you open Trailarr in browser, it will ask you to login. Default login is:
        
        ```bash
        Username: admin
        Password: trailarr
        ```

2. Go to `Settings` > `Connections` and Click the `Add New` button to add your Radarr and Sonarr instances.
![Add New](../getting-started/add-new.png)

3. Add your Radarr and Sonarr instances.
![Add Connection](add-connection.png)

4. Set the `Connection Name` to a name of your choice.

5. Set the `Type` to either `Radarr` or `Sonarr`.

6. Set the `Monitor Type` to your preference. See [Monitor Types](#monitor-types) below for more information.

    !!! tip
        _You can set different monitor types for each Radarr/Sonarr instance._


7. Add the `URL` for your Radarr or Sonarr instance.
    Full URL of your Radarr or Sonarr instance including the port number. For example, `http://192.168.0.15:6969`.

8. Add the `API Key` from your Radarr or Sonarr instance.
    Get `API Key` by opening Radarr/Sonarr in your browser, going to `Settings > General`, then copy the API key.

9. If you have a setup that requires `Path Mappings` to be set as described in [Volume Mapping](../getting-started/volume-mapping.md), you can add them here. See [Path Mapping](#path-mapping) below for more information.

    If you need additional help, there is a [Docker Config Tool](https://nandyalu.github.io/trailarr/usage/docker-builder.html){:target="_blank"} that will help you generate the docker-compose file with the correct volume mappings and path mappings.

    !!! tip
        _You can set different path mappings for each Radarr/Sonarr instance._

    !!! warning
        _Path mappings are optional and only required if you have a setup that requires them._

10. Click the `Save` button to save the connection.

11. Repeat the steps for each Radarr and Sonarr instance you want to monitor.

12. That's it! The application will now start downloading trailers for your media library. See [settings](../getting-started/application-settings.md) for more information on how to adjust settings.

## Monitor Types

### `Missing`
Monitors and downloads trailers for movies/series without a trailer.

### `New`
Only Monitors and download trailers for movies/series that gets added after the change.

### `Sync`
Monitors and downloads trailers for movie/series that are monitored in Radarr/Sonarr.

### `None`
Turns off monitoring for the connection and does not download any trailers.

!!! tip
    _If you have a huge library and don't want to download trailers for all of them, set the monitor type to `None` when adding a Radarr/Sonarr Connection. Wait for an hour or so to let the app sync all media from that connection, and change it to `New` to download trailers for new media. You can always manually set the monitor type for the movies/series you want to download trailers for._

## Path Mapping

In simple words, Path Mappings are used to tell Trailarr what to replace the path with when it gets a path from Radarr/Sonarr. `Path From` is what it looks for in the path, and `Path To` is what it replaces it with.

If you are using Windows, you might need to add path mappings in most cases, unless you are also running Radarr/Sonarr as docker containers. See [Volume Mapping](../install/volume-mapping.md) for more information.

!!! info
    Path mappings are optional and only required if you have a setup that requires them. You can set different path mappings for each Radarr/Sonarr instance.

??? example
    Adding a path mapping for a connection as:
    ```yaml
    Path From: C:\Users\username\Movies
    Path To: /media/movies
    ```
    will replace `C:\Users\username\Movies` with `/media/movies` in the path received from Radarr/Sonarr.
    So, if Radarr/Sonarr sends a path like `C:\Users\username\Movies\The MAtrix (1999)`, it will be replaced with `/media/movies/The Matrix (1999)`.

1. Click the `Add Path Mapping` button.
![Add Path Mapping](../getting-started/add-path-mapping.png)

2. Set the `Path From` to the path inside Radarr/Sonarr connection.

3. Set the `Path To` to the path inside the Trailarr container.
![Path Mapping](../getting-started/path-mapping.png)

4. Click the `Save` button to save the path mapping.

5. Repeat the steps for each path mapping you want to add, if needed.

!!! tip
    _Path mappings are useful when the media folder available to Radarr/Sonarr is same for multiple connections. For example, if Radarr has media folder set to `/media` and Sonarr has media folder set to `/media`, you can map the local folder for Radarr media to `/media/movies` and Sonarr media to `/media/tv` and then in add path mappings with `Path From` set to `/media` and `Path To` set to `/media/movies` for Radarr and `/media/tv` for Sonarr connections._

!!! warning
    For Path Mappings to work, you need to set the `Path From` to the exact path inside Radarr/Sonarr connection and `Path To` to the exact path inside the Trailarr container. If the paths do not match, the path mapping will not work. Path Mappings needs to be paired with `Volume Mapping` in the `docker-compose.yml` file. See [Volume Mapping](../install/volume-mapping.md) for more information.