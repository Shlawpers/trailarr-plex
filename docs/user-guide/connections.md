ğŸš§ Under Construction... ğŸš§
To be Updated . . .