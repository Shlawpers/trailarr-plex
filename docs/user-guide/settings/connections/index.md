# Connections

## Edit Connection Fields

- **Name**: A unique name for your connection (min. 3 characters).
- **Arr Type**: Select either Radarr or Sonarr.
- **Monitor Type**: Choose how trailers are monitored/downloaded (see below).
- **Server URL**: The full URL to your Radarr/Sonarr server (e.g., http://192.168.0.15:7878).
- **API Key**: The API key from your Radarr/Sonarr server settings.
- **Path Mappings**: Map Radarr/Sonarr internal paths to Trailarr paths for correct file access.

### Monitor Types

#### `Missing`
Monitors and downloads trailers for movies/series without a trailer.

#### `New`
Only Monitors and download trailers for movies/series that gets added after the change.

#### `Sync`
Monitors and downloads trailers for movie/series that are monitored in Radarr/Sonarr.

#### `None`
Turns off monitoring for the connection and does not download any trailers.

!!! tip
    _If you have a huge library and don't want to download trailers for all of them, set the monitor type to `None` when adding a Radarr/Sonarr Connection. Wait for an hour or so to let the app sync all media from that connection, and change it to `New` to download trailers for new media. You can always manually set the monitor type for the movies/series you want to download trailers for._