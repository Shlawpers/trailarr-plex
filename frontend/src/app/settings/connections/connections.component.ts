import { Component } from '@angular/core';
import { RouterOutlet } from '@angular/router';

@Component({
    selector: 'app-connections',
    imports: [RouterOutlet],
    templateUrl: './connections.component.html',
    styleUrl: './connections.component.css'
})
export class ConnectionsComponent {

}