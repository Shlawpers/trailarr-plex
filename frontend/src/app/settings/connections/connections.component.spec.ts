import {MockBuilder, MockedComponentFixture, MockRender} from 'ng-mocks';
import {ConnectionsComponent} from './connections.component';

describe('ConnectionsComponent', () => {
  let fixture: MockedComponentFixture<ConnectionsComponent>;

  beforeEach(() => MockBuilder(ConnectionsComponent));

  beforeEach(() => {
    fixture = MockRender(ConnectionsComponent);
    fixture.detectChanges();
  });

  it('has instance', () => expect(fixture.componentInstance).toBeTruthy());

  it('renders', () => expect(fixture).toMatchSnapshot());
});
