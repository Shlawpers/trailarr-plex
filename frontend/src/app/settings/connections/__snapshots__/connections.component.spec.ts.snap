// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionsComponent renders 1`] = `
<mock-render>
  <app-connections>
    <router-outlet />
  </app-connections>
</mock-render>
`;
